loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,362 @@
+# Summary
+This document acts as a Request for Comments (RFC) on the topic of how a Quilt mod's metadata should be formatted. Such changes are intended to provide added value for mod developers, library developers, and users alike.
+
+# Motivation
+Since the toolchain is being forked, now is the opportune moment to make large scale changes to many of the systems mod developers use. This RFC describes a more flexible and powerful format for the mod file that defines what a mods definition should look like.
+
+# Explanation
+Below is an outline of all defined keys and values.
+
+* [schema_version](#the-schema_version-field) — The schemaVersion to be used for reading this file
+* [quilt_loader](#the-quilt_loader-field) — Information related to loading the mod
+    * [group_id](#the-group_id-field) — The Maven group id
+    * [mod_id](#the-mod_id-field) — The mod id
+    * [provides](#the-provides-field) — Alternative mod ids provided by this mod
+    * [version](#the-version-field) — The mods version
+    * [entrypoints](#the-entrypoints-field) — Collection of entrypoints
+    * [plugins](#the-plugins-field) — Collection of plugins
+    * [jars](#the-jars-field) — Array of nested JARs to be loaded
+    * [language_adapters](#the-language_adapters-field) — Array of language adapters
+    * [depends](#the-depends-field) — Collection of mod dependencies
+    * [breaks](#the-breaks-field) — Collection of mods that this mod is incompatible with
```

Comment:
Why are the dependency resolution types limited to `depends` and `breaks`?  
More types of the mod relations could be added and all of them could be grouped in a `relation` object.

Reply:
Just for some context - not asserting a decision here: the old `suggests` resolution type was never really used. `conflicts` type did print a warning but it was always really a soft resolution type.